mon-pix/tests/unit/mixins/secured-route-mixin-test.js

```diff
@@ -0,0 +1,166 @@
+      this.owner.register('service:currentUser', Service.extend({
+        user: {}
+      }));
+
+      route = Route.extend(MixinImplementingBeforeModel, SecuredRouteMixin).create();
+      setOwner(route, this.owner);
+
+      sinon.spy(transition, 'send');
+      sinon.spy(router, 'transitionTo');
+    });
+
+    describe('when the session is authenticated', function() {
+
+      beforeEach(function() {
+        const session = this.owner.lookup('service:session');
+        session.set('isAuthenticated', true);
+      });
+
+      describe('when the user has accepted the terms of service', function() {
+
+        beforeEach(function() {
+          const currentUser = this.owner.lookup('service:currentUser');
+          currentUser.user.mustValidateTermsOfService = false;
+        });
+
+        it('returns the upstream promise', async function() {
+          const result = await route.beforeModel(transition);
+
+          expect(result).to.equal('upstreamReturnValue');
+        });
+
+        it('does not transition to the authentication route', function() {
+          route.beforeModel(transition);
+
+          sinon.assert.neverCalledWith(router.transitionTo, 'login');
+        });
+
+        it('does not transition to the terms-of-service route', function() {
+          route.beforeModel(transition);
+
+          sinon.assert.neverCalledWith(router.transitionTo, 'terms-of-service');
+        });
+      });
+
+      describe('when the user has not accepted the terms of service', function() {
+
+        beforeEach(function() {
+          const currentUser = this.owner.lookup('service:currentUser');
+          currentUser.user.mustValidateTermsOfService = true;
+        });
+
+        it('does not returns the upstream promise', async function() {
+          expect(route.beforeModel(transition)).to.be.undefined;
+        });
+
+        it('does not transition to the authentication route', function() {
+          route.beforeModel(transition);
+
+          sinon.assert.neverCalledWith(router.transitionTo, 'login');
+        });
+
+        it('does transition to the terms-of-service route', function() {
+          route.beforeModel(transition);
+
+          sinon.assert.calledWith(router.transitionTo, 'terms-of-service');
+        });
+      });
+    });
+
+    describe('when the session is not authenticated', function() {
+
+      it('does not return the upstream promise', function() {
+        expect(route.beforeModel(transition)).to.be.undefined;
+      });
+
+      it('transitions to "login" as the default authentication route', function() {
+        route.beforeModel(transition);
+        sinon.assert.calledWith(router.transitionTo, 'login');
+      });
+
+      it('does not transition to the terms-of-service route', function() {
+        route.beforeModel(transition);
+        sinon.assert.neverCalledWith(router.transitionTo, 'terms-of-service');
+      });
+
+      it('transitions to the set authentication route', function() {
```

Comment:
```suggestion
      it('transitions to the authentication route', function() {
```
Ca évoque l'intention au lieu du détail d'implémentation